arelle/plugin/validate/NL/PluginValidationDataExtension.py

```diff
@@ -80,6 +84,24 @@ def checkContexts(self, allContexts: dict[str, list[ModelContext]]) -> None:
         self._contextsWithPeriodTimeZone = contextsWithPeriodTimeZone
         self._contextsWithSegments = contextsWithSegments
 
+    def checkFootnote(self, modelXbrl: ModelXbrl) -> None:
+        factLangs = self.factLangs(modelXbrl)
+        footnotesRelationshipSet = modelXbrl.relationshipSet("XBRL-footnotes")
+        orphanedFootnotes = set()
+        noMatchLangFootnotes = set()
+        for elts in modelXbrl.ixdsEltById.values():   # type: ignore[attr-defined]
+            for uncastElt in elts:
+                elt = cast(Any, uncastElt)
```

Comment:
Out of curiosity, what does this line do / why is it required?  I would have guessed that the `isinstance` on the next line would have made this line unnecessary.

Reply:
It is there as a relic of a typing issue I had been having.  Nice catch!